types/tx.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import "math/big"
+
+// TxType is the transaction type.
+type TxType byte
+
+// List of supported transaction types
+const (
+	LegacyTx     TxType = 0x0
+	StateTx      TxType = 0x7f
+	DynamicGeeTx TxType = 0x8f
+)
+
+// String returns string representation of the transaction type.
+func (t TxType) String() (s string) {
+	switch t {
+	case LegacyTx:
+		return "LegacyTx"
+	case StateTx:
+		return "StateTx"
+	case DynamicGeeTx:
+		return "DynamicGeeTx"
+	default:
+		return "UnknownTX"
+	}
+}
+
+// Tx represents the general transaction behavior interface.
+// It contains common transaction field getters and setters as
+// well as additional functions for managing transaction.
+// Any modification function should return modified transaction object.
+type Tx interface {
+	// Nonce returns transaction nonce.
+	Nonce() uint64
+
+	// SetNonce sets the given transaction nonce.
+	SetNonce(nonce uint64) Tx
+
+	// GasPrice returns transaction gas price.
+	GasPrice() *big.Int
+
+	// SetGasPrice sets the given transaction gas price.
+	SetGasPrice(gasPrice *big.Int) Tx
+
+	// Gas returns transaction gas.
+	Gas() uint64
+
+	// SetGas sets the given transaction gas.
+	SetGas(gas uint64) Tx
+
+	// To returns transaction receiver address.
+	// It could be nil in case of the state transaction.
+	To() *Address
+
+	// SetTo sets the given transaction receiver address.
+	SetTo(addr Address) Tx
```

Comment:
I'm trying to finish reviews for all rfc/pr's that required my review in december. It is my tech dept, because i was overloaded and exhausted in december.